Fix #3222: Add ability to zoom pages

Brandon-T opened the pull request:
## Summary of Changes
- Added global zoom level.
- Allow site level preference to take priority over global preference
- Update pages zoom-levels when Zoom changes from the UI in real-time.

<!-- Enter a ticket number for this PR, create a new one if it is not there yet. -->
This pull request fixes #3222

## Screenshots:

<img width="348" alt="image" src="https://user-images.githubusercontent.com/1530031/164571000-ea7101fd-a7a1-45d5-b30d-69246fe0a204.png"> <img width="348" alt="image" src="https://user-images.githubusercontent.com/1530031/164571083-eec35f17-5ba6-491d-9be7-81379076dc77.png"> <img width="348" alt="image" src="https://user-images.githubusercontent.com/1530031/164571106-144ff8ed-b8ca-4896-b373-befce12ff4e6.png"> <img width="348" alt="image" src="https://user-images.githubusercontent.com/1530031/164571173-52366ce1-53a1-4e32-b090-af05645ed71b.png">



## Submitter Checklist:

- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`


## Reviewer Checklist:

- [x] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `bug` / `enhancement`
- [x] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [x] Adequate unit test coverage exists to prevent regressions.
- [x] Adequate test plan exists for QA to validate (if applicable).
- [x] Issue and pull request is assigned to a milestone (should happen at merge time).

iccub reviewed (changes requested):
Few DB questions, will do functional review later

Brandon-T committed:
Added Page Zoom
Localized strings. Added empty zoom screen strings.
Added global zoom level.
Allow site level preference to take priority over global preference

Much better improvements to the UI of the zoom-bar.
Huge accessibility improvements using contentShape to allow the entire shape of the buttons to be tapped.

Moved overlay to its own variable. Adjust font on invisible view.

Show zoom level if set, no matter what.
Make zoom levels destructive.
Normalize Zoom levels on exit.
Add custom notification for when zoom level changes.

Reset should be default value, and not 1.0.

Disable Zoom Settings in Private-Browsing, but allow zooming a website individually

Changed font and colours.
Add comments and disabled accentColor
Fixed filtering.

Use subheadline for section headers.

iccub reviewed (approved):
🔥
